packages/perseus/src/widgets/interactive-graphs/reducer/interactive-graph-reducer.ts

```diff
@@ -499,6 +503,31 @@ function doChangeRange(
     };
 }
 
+function doAddPoint(
+    state: InteractiveGraphState,
+    action: AddPoint,
+): InteractiveGraphState {
+    if (state.type !== "point") {
+        return state;
+    }
+    const {snapStep} = state;
+    const snappedPoint = snap(snapStep, action.location);
+
+    // Check if there's already a point in that spot
+    for (const point of state.coords) {
+        if (point[0] === snappedPoint[0] && point[1] === snappedPoint[1]) {
```

Comment:
We have constants for the X and Y components of a vector, which might make this more self-explanatory:
```suggestion
        if (point[X] === snappedPoint[X] && point[Y] === snappedPoint[Y]) {
```